core/plugins/org.polarsys.capella.core.af.integration/src/org/polarsys/capella/core/af/integration/CapellaMetadataProvider.java

```diff
@@ -1,261 +1,269 @@
+
+    ResourceSet set = new ResourceSetImpl();
+    set.getLoadOptions().put(GMFResource.OPTION_ABORT_ON_ERROR, Boolean.TRUE);
+    set.getLoadOptions().put(XMLResource.OPTION_RECORD_UNKNOWN_FEATURE, Boolean.TRUE);
+    try {
+      URI uri = EcoreUtil2.getURI(file);
+      set.getResource(uri, true);
+      return checkMetadata(uri, set);
+
+    } catch (Exception e) {
+      return new Status(IStatus.ERROR, AFIntegrationPlugin.getSymbolicName(), e.getMessage());
+
+    } finally {
+      for (Resource r : set.getResources()) {
+        r.unload();
+      }
+      set.getResources().clear();
+    }
+  }
+
+  /**
+   * For a given AFM file loaded in a resourceSet, retrieve if the model is fully compatible with the current platform.
+   * Checks current platform version and viewpoints.
+   */
+  public IStatus checkMetadata(URI sessionResourceURI, ResourceSet set) {
+    if (!ViewpointManager.getInstance(set).hasMetadata()) {
+      // If there is no afm file but the current check is about a capella project, we must raise an exception because
+      // the afm file is missing.
+      if (CapellaResourceHelper.isCapellaProject(sessionResourceURI)) {
+        return new Status(IStatus.ERROR, AFIntegrationPlugin.getSymbolicName(),
+            NLS.bind(Messages.NoMetadataException_Message,
+                MetadataHelper.getViewpointMetadata(set).getExpectedMetadataStorageURI().toPlatformString(true)));
+      }
+    } else {
+      // If there is an afm file, first we check if the capella viewpoint is present in the platform
+      org.polarsys.kitalpha.resourcereuse.model.Resource vp = ViewpointManager
+          .getViewpoint(AFIntegrationPlugin.CAPELLA_VIEWPOINT_ID);
+      if (vp != null) {
+        // then, if the afm file has this capella viewpoint
+        Version fileVersion = MetadataHelper.getViewpointMetadata(set).getVersion(vp);
+        if (fileVersion != null) {
+          Version currentVersion = getCurrentVersion();
+
+          // If version of Capella is not compatible (not same major/minor), we raise a migration required message
+          IStatus migration = isMigrationRequired(fileVersion, currentVersion);
+          if (!migration.isOK()) {
+            return migration;
+          }
+        }
+      }
+
+      // If this viewpoint or another viewpoint is not compatible (not same major/minor), we raise a
+      // missing/incompatible viewpoint
+      IStatus result = ViewpointManager.checkViewpointsCompliancy(set);
+      if (!result.isOK()) {
+        return result;
+      }
+    }
+
+    return Status.OK_STATUS;
+  }
+
+  /**
+   * Returns whether the file of the given fileVersion requires a migration to be compatible to currentVersion
+   * 
+   * @see See also org.polarsys.capella.core.data.migration.af.ViewpointMigrationContribution.isMigrationPossible
+   */
+  public IStatus isMigrationRequired(Version fileVersion, Version currentVersion) {
+
+    // If model from <MAJOR>.x towards <MAJOR>.y, we requires a migration.

```

Comment:
Maybe just this comment should say that we explicitly need a migration from 7.0.x to 7.0.y